### ISSUE TITLE
%confdir not set in rc.conf

### ISSUE THREAD

**Issue by SilverFishCat** [ranger/ranger#1466]
#### Runtime Environment

- Operating system and version: 4.20.7-arch1-1-ARCH
- Terminal emulator and version: alacritty 0.2.7
- Python version: 3.7.2 (default, Jan 10 2019, 23:51:51) [GCC 8.2.1 20181127]
- Ranger version/commit: ranger 1.9.2
- Locale: en_US.UTF-8

#### Current Behavior
When %confdir is used in rc.conf an error is printed to stdout.
The error is:
```
Error in line `echo %confdir':
  'NoneType' object has no attribute 'thisfile'
```
Ranger continous normally without executing that line

#### Expected Behavior
The %confdir macro should expand to $XDG_CONFIG_HOME/ranger or whatever else was given by --confdir

#### Context
I want to use source %confdir/<extraconf> to be able to change a set of key bindings with as little changes to rc.conf as possible (specifically, changing J and K), so these remapping can be shared easily.

#### Steps to reproduce
1. Add `echo %confdir` to rc.conf
2. Run ranger
3. Close ranger

**Comment by toonn** [comment]
I didn't think sourcing works in `rc.conf`?

**Comment by SilverFishCat** [comment]
Seems to work just fine for me.
I have the default rc.conf (or at least I didn't change anything that I can remember) with the line `source ~/.config/ranger/sometext` at the end.
That file only contains these lines:
```
copymap <UP>       j
copymap <DOWN>     k
```
This works, and the longer file works too as long as I don't use %confdir.

**Comment by toonn** [comment]
@Vifon, did you implement sourcing of configs? I thought we didn't have this?

**Comment by Vifon** [comment]
Huh? No, I don't think so.

**Comment by toonn** [comment]
@SilverFishCat, wouldn't `.` work like `%confdir` actually?

**Comment by SilverFishCat** [comment]
@toonn no, unless I run ranger with `~/.config/ranger/` as my working directory

**Comment by toonn** [comment]
I'd expect relative paths in `rc.conf` to be relative to its location tbh.

`%confdir` probably doesn't work because `rc.conf` is executed very early on startup.

**Comment by palikar** [comment]
Ok, I've looked into this issue and I've figured it out.

@toonn  sourcing __is__ supported in `rc.conf` as `source` is a function defined in `action.py`. Everything in `actions.py` can be used as a console command or command in `rc.conf`. The sourcing is not the problem here.
The actual issue is that when expanding macros, the function (`get_macros` in `actions.py`) expects that the FM object is fully initialized and `thistab` and `thisdir` properties are there and everything. In the time of sourcing the rc.conf, however, there are set to None. So this is whats causing the 
```
Error in line `echo %confdir':
  'NoneType' object has no attribute 'thisfile'
```
as `thisfile` relies on `thistab`.

I've added some additional checks in some places and I've got it to work now. I'll make a PR later this day and I am pretty sure I've solved the bug and the `%confdir` macro will be usable in the 'rc.conf' as well as sourcing based on it.

**Comment by toonn** [comment]
That's exactly what I said. `rc.conf` is executed or sourced or whatever you want to call it too early.
